### PR TITLE
Morse output synced with input

### PR DESCRIPTION
**Pull request by jirik**

### CODE REVIEW

File: pages/morse.js [hunk 1]
```diff
@@ -51,7 +51,25 @@ const allResults = (message) => {
...
+  const inputItems = []
+
+  decode(message).map((msgPart) => {
+    inputItems.push({
+      input: msgPart.string[0],
+      output: {
+        char: (msgPart.char || '').toUpperCase() || null,
+      },
+    })
+    msgPart.string
```
Comment (index-git) [inline pages/morse.js:79]: For me inputItems.concate seems more readable than `push` inside `map`.

Comment (jirik) [reply pages/morse.js:79]: I think `map` is wrong anyway, `forEach` would be better.
I am not sure how exactly you imagine the code with `concat`.

Comment (index-git) [reply pages/morse.js:79]: `forEach` sounds good.

File: pages/morse.js [hunk 2]
```diff
@@ -51,7 +51,25 @@ const allResults = (message) => {
 }
 
 const messageToBox = (message) => {
-  return <MorseResultBox label='Základní řešení' message={message} />
+  const inputItems = []
+
+  decode(message).map((msgPart) => {
```
Comment (index-git) [inline pages/morse.js:56]: I suggest to separate `decode` command and add unit tests to it.

File: component/MorseResultBox.js [hunk 3]
```diff
@@ -3,13 +3,44 @@ import * as React from 'react'
 import Box from '@mui/material/Box'
 import morse_styles from '../styles/morse.module.scss'
 
+const getJoinerClass = (showJoiner, isStartItem, isEndItem) => {
```
Comment (index-git) [inline component/MorseResultBox.js:11]: Add unit tests.

**Review by index-git — CHANGES_REQUESTED** [review]

File: pages/morse.js [hunk 4]
```diff
@@ -51,7 +51,25 @@ const allResults = (message) => {
...
+    inputItems.push({
+      input: msgPart.string[0],
+      output: {
+        char: (msgPart.char || '').toUpperCase() || null,
+      },
+    })
+    msgPart.string
+      .split('')
+      .slice(1)
+      .map((msgChar) => {
```
Comment (jirik) [inline pages/morse.js:82]: `map` -> `forEach` or consider map with return

**Review by index-git — APPROVED** [review]

**Pull request merged**